src/ophyd_async/core/signal.py

```diff
@@ -557,29 +562,77 @@ async def wait_for_value(
     await checker.wait_for_value(signal, timeout)
 
 
+async def set_and_wait_for_other_value(
+    set_signal: SignalRW[T],
+    set_value: T,
+    read_signal: SignalR[S],
+    read_value: S,
+    timeout: float = DEFAULT_TIMEOUT,
+    set_timeout: Optional[float] = None,
+) -> AsyncStatus:
+    """Set a signal and monitor another signal until it has the specified value.
+
+    This function sets a set_signal to a specified set_value and waits for
+    a read_signal to have the read_value.
+
+    Parameters
+    ----------
+    set_signal:
+        The signal to set
+    set_value:
+        The value to set it to
+    read_signal:
+        The signal to monitor
+    read_value:
+        The value to wait for
+    timeout:
+        How long to wait for the signal to have the value
+    set_timeout:
+        How long to wait for the set to complete
+
+    Notes
+    -----
+    Example usage::
+
+        set_and_wait_for_value(device.acquire, 1, device.acquire_rbv, 1)
+    """
+    # Start monitoring before the set to avoid a race condition
+    values_gen = observe_value(read_signal)
```

Comment:
There is still a tiny race condition here. What we are effectively doing for a CA signal is:
- Create a task that will eventually call `ca_create_subscription`
- Create a task that will eventually call `ca_array_put_callback`
- Wait until a subscription update matches

It is possible (but less likely) that we might send a put which might send an update before the first subscription connects. A possible fix is to consume the first value from `observe_value` before doing `set`, so we are sure we have a live subscription.